pkg/network/dns/cache.go

```diff
@@ -3,30 +3,38 @@
 // This product includes software developed at Datadog (https://www.datadoghq.com/).
 // Copyright 2016-present Datadog, Inc.
 
-//go:build windows || linux_bpf
-// +build windows linux_bpf
+//go:build (windows && npm) || linux_bpf
+// +build windows,npm linux_bpf
 
 package dns
 
 import (
 	"sync"
 	"time"
 
-	"go.uber.org/atomic"
-
+	nettelemetry "github.com/DataDog/datadog-agent/pkg/network/telemetry"
 	"github.com/DataDog/datadog-agent/pkg/process/util"
 	"github.com/DataDog/datadog-agent/pkg/util/log"
 )
 
-type reverseDNSCache struct {
-	// Telemetry
-	length    *atomic.Int64
-	lookups   *atomic.Int64
-	resolved  *atomic.Int64
-	added     *atomic.Int64
-	expired   *atomic.Int64
-	oversized *atomic.Int64
+// Telemetry
+var cacheTelemetry = struct {
```

Comment:
This is a nitpicky thing, but this global will cause these objects to be allocated if this package is imported, regardless if you ever even use the `reverseDNSCache` object. It might be better to use `sync.Once` to create these on-demand once.

Reply:
It seems like nobody else is doing this, so maybe this is an over-optimization.

Reply:
A nice pattern could be to embed this in the `reverseDNSCache` struct. A nice side effect would be we'd have a clean way to test telemetry if we wanted to.

```
type cacheTelemetry struct {
	field int
	stats struct {
		cacheSize nettelemetry.StatGaugeWrapper
	}
}
```


Then use
```
cacheTelemetry.stats.cacheSize.Get()
```

In a test